pkg/vault/test/foundry/RouterCommon.t.sol

```diff
@@ -89,26 +89,37 @@ contract RouterCommonTest is BaseVaultTest {
         _fillEthStateTestAfter(vars);
 
         assertEq(vars.bobEthAfter, vars.bobEthBefore, "Bob ETH balance is wrong");
-        assertEq(vars.vaultEthAfter, vars.vaultEthBefore, "Vault ETH balance is wrong");
+        assertEq(vars.vaultEthBefore, 0, "Vault had ETH balance before");
+        assertEq(vars.vaultEthAfter, 0, "Vault has ETH balance after");
         assertEq(vars.bobWethAfter, vars.bobWethBefore - amountToDeposit, "Bob WETH balance is wrong");
         assertEq(vars.vaultWethAfter, vars.vaultWethBefore + amountToDeposit, "Vault WETH balance is wrong");
+        assertEq(vars.wethDeltaAfter, vars.wethDeltaBefore - int256(amountToDeposit), "Vault delta is wrong");
     }
 
     function testSendTokenOutWethIsEth() public {
         vault.manualSetIsUnlocked(true);
+        vm.startPrank(lp);
+        uint256 wethDeposit = lp.balance / 10;
+        weth.deposit{ value: wethDeposit }();
+        weth.transfer(address(vault), wethDeposit);
+        vault.settle(weth, wethDeposit);
+        vm.stopPrank();
```

Comment:
Got it!

Reply:
This is just a mock; you wouldn't do it that way in practice. But since we only care about balance differences and token delta differences it's fine for testing purposes.

In practice this wouldn't settle.